src/test/java/br/com/fiap/techchallenge/adapters/GetPedidoAdapterTest.java

```diff
@@ -0,0 +1,85 @@
+package br.com.fiap.techchallenge.adapters;
+
+import br.com.fiap.techchallenge.adapters.pedido.GetPedidoAdapter;
+import br.com.fiap.techchallenge.domain.entities.Pedido;
+import br.com.fiap.techchallenge.domain.model.mapper.pedido.PedidoMapper;
+import br.com.fiap.techchallenge.domain.valueobjects.PedidoDTO;
+import br.com.fiap.techchallenge.infra.repositories.PedidoRepository;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.InjectMocks;
+import org.mockito.Mock;
+import org.mockito.MockitoAnnotations;
+import org.springframework.data.domain.Page;
+import org.springframework.data.domain.PageImpl;
+import org.springframework.data.domain.PageRequest;
+import java.util.Arrays;
+import java.util.List;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.when;
+
+public class GetPedidoAdapterTest {
+
+    @Mock
+    private PedidoRepository pedidoRepository;
+
+    @Mock
+    private PedidoMapper pedidoMapper;
+
+    @InjectMocks
+    private GetPedidoAdapter getPedidoAdapter;
+
+    @BeforeEach
+    public void setup() {
+        MockitoAnnotations.openMocks(this);
+    }
+
+    @Test
+    public void buscarPedidoPorId_ReturnsPedidoDTO_WhenPedidoExists() {
```

Comment:
Os testes devem sempre começar usando o padrão do TDD:
should -> Para cenários de sucesso
must -> Para cenários de falha

Os critérios do teste não devem ser separados por underscore (_)
Os critérios do teste devem ser em pt-BR.